src/main/java/com/hcl/appscan/jenkins/plugin/scanners/Scanner.java

```diff
@@ -76,4 +66,36 @@ protected String resolvePath(String path, VariableResolver<String> resolver) {
 
 		return path;
 	}
+    protected void validateGeneralSettings(JenkinsAuthenticationProvider authProvider, Map<String, String> properties, IProgress progress, boolean isAppScan360) throws IOException {
+        if (isAppScan360) {
+            if (!properties.get("FullyAutomatic").equals("true")) {
+                progress.setStatus(new Message(Message.WARNING, Messages.warning_allow_intervention_AppScan360()));
+            }
+        } else if (authProvider.getacceptInvalidCerts()) {
+            progress.setStatus(new Message(Message.WARNING, Messages.warning_asoc_certificates()));
+        }
+
+        if(properties.containsKey(CoreConstants.SCAN_ID)) {
+            if(properties.get(CoreConstants.PERSONAL_SCAN).equals("true")) {
+                progress.setStatus(new Message(Message.WARNING, Messages.warning_personal_scan_rescan()));
+            }
+        }
+    }
+
+    protected void scanIdValidation(JSONObject scanDetails, Map<String, String> properties) throws JSONException, IOException {
```

Comment:
Similar to my other comment, we shouldn't force the subclasses to know that they need to call scanIdValidation(). This should be a private method that's called in the Scanner.validateSettings() method. Also, the name should be an action, like validateScanId().

Reply:
Okay, Now, calling the validateScanId() method call from the generic validateSettings() method.